### PR TITLE
Update docs in preparation for replacing Match.jl

### PR DESCRIPTION
**Pull request by gafter**

### CODE REVIEW

**Comment by gafter** [comment]
@mcmcgrath13 Could you please have a look at these changes?

File: src/documentation.jl [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+```
+"""
+macro match end
+
+"""
+    @match_return value
+
+Inside the result part of a @match case, you can return a given value early.
+
+# Examples
```
Comment (mcmcgrath13) [inline src/documentation.jl:131]: a `documentation.jl` file isn't something I've come across before. Typically folks add the docstrings on the regular declaration site (either in a themed file, or in the main `Package.jl` file

File: src/macros.jl [hunk 2]
```diff
@@ -85,103 +87,44 @@ Usage:
...
+#         pattern1 => result1
+#         pattern2 => result2
+#         ...
+#     end
+# ```
+
+# Return `result` for the first matching `pattern`.
+# If there are no matches, throw `MatchFailure`.
+# """
+#
```
Comment (mcmcgrath13) [inline src/macros.jl:112]: I would bring back the proper docstrings here

File: test/rematch2.jl [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 file = Symbol(@__FILE__)
 
-@enum Color Yellow Green Blue
+@enum Color Yellow Blue Greed
```
Comment (mcmcgrath13) [inline test/rematch2.jl:9]:
```suggestion
@enum Color Yellow Blue Green
```

Comment (mcmcgrath13) [reply test/rematch2.jl:9]: lol - I see it now

File: src/documentation.jl (hunk 1 above)
Comment (gafter) [reply src/documentation.jl:131]: Unfortunately, there are several places where different versions of the macros are defined, and the documentation just gets concatenated when you do that.  This ensures we have a single documentation form for each name.

File: src/macros.jl (hunk 2 above)
Comment (gafter) [reply src/macros.jl:112]: See above.  I could put them in this file, but not on every declaration.

**Review by mcmcgrath13 — APPROVED** [review]
thanks!

**Pull request merged**